shell/browser/api/electron_api_in_app_purchase.cc

```diff
@@ -23,6 +39,12 @@ struct Converter<in_app_purchase::Payment> {
     dict.SetHidden("simple", true);
     dict.Set("productIdentifier", payment.productIdentifier);
     dict.Set("quantity", payment.quantity);
+    if (!payment.applicationUsername.empty()) {
+      dict.Set("applicationUsername", payment.applicationUsername);
+    }
+    if (!payment.paymentDiscount.identifier.empty()) {
+      dict.Set("paymentDiscount", payment.paymentDiscount);
+    }
```

Comment:
Not quite sure whether this is the right way to set an `undefined` field.
I tried setting it in Objective C code, but it doesn't seem to work.

Reply:
Hmmm, ok, it looks like the other fields that belong here also do the same thing. 
I have removed all optional for now to make them consistent.